craft_parts/plugins/autotools_plugin.py

```diff
@@ -0,0 +1,95 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2020 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The autotools plugin implementation."""
+
+from typing import Any, Dict, List, Set, cast
+
+from .base import Plugin, PluginModel, extract_plugin_properties
+from .properties import PluginProperties
+
+
+class AutotoolsPluginProperties(PluginProperties, PluginModel):
+    """The part properties used by the make plugin."""
+
+    autotools_configure_parameters: List[str] = []
+
+    @classmethod
+    def unmarshal(cls, data: Dict[str, Any]):
+        """Populate dump properties from the part specification.
+
+        :param data: A dictionary containing part properties.
+
+        :return: The populated plugin properties data object.
+
+        :raise pydantic.ValidationError: If validation fails.
+        """
+        plugin_data = extract_plugin_properties(data, plugin_name="autotools")
+        return cls(**plugin_data)
+
+
+class AutotoolsPlugin(Plugin):
+    """The autotools plugin is used for autotools-based parts.
+
+    Autotools-based projects are the ones that have the usual
+    `./configure && make && make install` instruction set.
+
+    This plugin will check for the existence of a 'configure' file, if one
+    cannot be found, it will run 'autoconf --install'.
```

Comment:
```suggestion
    This plugin will check for the existence of a 'configure' file, if one
    cannot be found, it will first try to run 'autogen.sh' or 'bootstrap'
    to generate one.
```

Reply:
We should fix that in snapcraft as well.